pkg/ingester/ingester.go

```diff
@@ -263,6 +305,10 @@ func (i *Ingester) Shutdown() {
 		close(i.quit)
 		i.done.Wait()
 
+		if i.wal != nil {
```

Comment:
Can we make sure that the WAL is never nil? Like we have the `noop{}` WAL? I am not a fan of `if x != nil then x.close()`

Reply:
I think it was added because of some test, else it would never be nil in the actual runtime. I will check about that and remove this nil check.